### DISCUSSION TITLE
Importing existing jbox2d project

### DISCUSSION THREAD

**Discussion by adamkatav** [flame-engine/flame#1473] [Q&A] [answered]
Hello, I have a complete project for Jbox2d (in Java of course) and I would like to use it in flutter (for Android).
Is it possible to run the project (say Jbox testbed) with forge2d?
If so, how? And if not, how would you approach to recreating Jbox testbed in forge2d?
Thank you very much!

**Comment by spydon** [comment, marked as answer]
> Is it possible to run the project (say Jbox testbed) with forge2d?

No, you can't run the Jbox testbed in Forge2D, both since it is is not Dart and since that Dart port of JBox2D forked off from an old version of JBox2D.

> If so, how? And if not, how would you approach to recreating Jbox testbed in forge2d?

There is unfortunately no easy way to do this, you'll have to port all the code that you want over from the projects that you want to run to Forge2D.

**Reply by adamkatav** [reply]
Thank you very much for the quick answer, guess I'll have to port it :)

**Reply by adamkatav** [reply]
On that note, is it possible to implement a spring in flame?
It is the only thing I need to avoid using forge2d

**Reply by spydon** [reply]
You can use a `DistanceJoint` just like in normal box2d.
